### ISSUE TITLE
Refactor README document into one or more manuals

### ISSUE THREAD

**Issue by mojavelinux** [asciidoctor/asciidoctorj#333]
The README is no longer a README :) It has matured into a full-blown manual (> 64K). I think it's time to setup a docs folder and refactor the README into one or more manuals.

The README should be simplified to a simple introduction, basic setup procedures (“getting started”) and some general information about the project (license, mailing list, etc).

Ideally, we should have a file for each major topic. For instance, the instructions for creating a converter would be it's own file. Same for how to create an extension.

We can use this issue to outline the files and sections.

**Comment by mojavelinux** [comment]
There are several distinct user types, so perhaps several manuals.

* There are people that want to use AsciidoctorJ as a replacement for Asciidoctor (interested primarily in the distribution)
* There are people that want to embed AsciidoctorJ into their application using the API to invoke Asciidoctor (integrators)
* There are people that want to extend AsciiDoc and the Asciidoctor processor to give it more functionality (extension writers)

Where a person creating a custom converter falls is a bit fuzzy because they might be either an integrator doing customization or an extension writer. Hard to say.

**Comment by lordofthejars** [comment]
:+1:

**Comment by robertpanzer** [comment]
> Where a person creating a custom converter falls is a bit fuzzy because they might be either an integrator doing customization or an extension writer. Hard to say.

I think that's true but not a problem. 
It's like having 4 chapters for three different readers. 
Reader A is happy with chapter 1, B with 1, 2 & 4 and C with 1, 3 & 4.

**Comment by robertpanzer** [comment]
What do you think about writing dedicated manuals and thereby reduce the README. 
That means provide a separate documentation like:
- Writing converters for custom formats
- Writing custom extensions
And replace the content in the README with links to these documents?

Should the documents be located under asciidoctorj/docs?

**Comment by mojavelinux** [comment]
> What do you think about writing dedicated manuals

Agree. I think we should plan to write individual documents as though they were chapters in a user manual. The more focused, the better. We can then produce a complete manual by combining these chapters together.

> Should the documents be located under asciidoctorj/docs?

Yes. We don't yet have a defined structure yet across projects, so AsciidoctorJ will lead the way. There's no doubt that "docs" is clear and easy to find...and that's requirement 1.

**Comment by robertpanzer** [comment]
Integrators Guide created as part of #361
